File: python-pr/52/python/code-quality-performance/matrix_multiplier_22.py
# {fact rule=code-quality-performance@v1.0 defects=1}
def multiply_matrices(a, b):
    rows, cols = len(a), len(b[0])
    result = [[0] * cols for _ in range(rows)]
    for i in range(rows):
        for j in range(cols):
            for k in range(len(a[0])):
                result[i][j] += a[i][k] * b[k][j]
    return result
# {/fact}

File: js-pr/56/javascript/code-quality-performance/matrix_multiplier_22.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function multiplyMatrices(a, b) {
    let rows = a.length;
    let cols = b[0].length;
    let result = Array(rows).fill().map(() => Array(cols).fill(0));
    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < cols; j++) {
            for (let k = 0; k < a[0].length; k++) {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    return result;
}
// {/fact}